src/main/java/com/develhope/spring/vehicles/entity/Vehicle.java

```diff
@@ -49,7 +48,8 @@ public class Vehicle {
     @Enumerated(EnumType.STRING)
     private VehicleState vehicleState;
     @NonNull
-    private boolean purchased;
+    private boolean isSellable;
```

Comment:
mi viene però il dubbio del caso in cui il veicolo sia solo ordinabile, diventa macchinoso identificare quel tipo di veicoli perchè solo con questi stati dovremmo filtrarci i veicoli che danno false a isSellable e non hanno lo stato NOT_AVAILABLE; non sarebbe meglio inserire direttamente un bollano anche per isOrderable?

Reply:
Se un veicolo è NOT_AVAILABLE, è probabile che sia disponibile per cause x, per il resto isSellable comporta che appunto il veicolo sia in vendita e questo implica il fatto che quindi non possa essere affittabile (isRentable)